nerfstudio/scripts/downloads/eyeful_tower.py

```diff
@@ -0,0 +1,357 @@
+            else:
+                raise NotImplementedError("This shouldn't happen")
+
+            T = np.array(camera["T"]).T  # Data stored as column-major
+            T = np.linalg.inv(T)
+            T = T[[2, 0, 1, 3], :]
+            T[:, 1:3] *= -1
+            frame["transform_matrix"] = T.tolist()
+
+            frames.append(frame)
+
+        frames = sorted(frames, key=lambda f: f["file_path"])
+
+        output["frames"] = frames
+        output["train_filenames"] = split_filenames["train"]
+        output["val_filenames"] = split_filenames["test"]
+        return output
+
+    def subsample_nerfstudio_transforms(self, transforms: dict, n: int):
+        target = min(len(transforms["frames"]), n)
+        indices = np.round(np.linspace(0, len(transforms["frames"]) - 1, target)).astype(int)
+
+        frames = []
+        for i in indices:
+            frames.append(transforms["frames"][i])
+
+        output = copy.deepcopy(transforms)
+        output["frames"] = frames
+
+        # Remove the unused files from the splits
+        filenames = {f["file_path"] for f in frames}
+        for key in ["train_filenames", "val_filenames"]:
+            output[key] = sorted(list(set(transforms[key]) & filenames))
+
+        return output
+
+    def download(self, save_dir: Path):
+        if len(self.capture_name) == 0:
+            self.capture_name = ("riverview",)
+            print(
+                f"No capture specified, using {self.capture_name} by default.",
+                "Add `--help` to this command to see all available captures.",
+            )
+
+        if len(self.resolution_name) == 0:
+            self.resolution_name = ("jpeg_2k",)
+            print(
+                f"No resolution specified, using {self.resolution_name} by default.",
+                "Add `--help` to this command to see all available resolutions.",
+            )
+
+        captures = set()
+        for capture in self.capture_name:
+            if capture == "all":
+                captures.update([c for c in eyefultower_downloads if c != "all"])
+            else:
+                captures.add(capture)
+        captures = sorted(captures)
+        if len(captures) == 0:
+            print("WARNING: No EyefulTower captures specified. Nothing will be downloaded.")
+
+        resolutions = set()
+        for resolution in self.resolution_name:
+            if resolution == "all":
+                resolutions.update([r for r in eyefultower_resolutions.keys() if r != "all"])
+            else:
+                resolutions.add(resolution)
+        resolutions = sorted(resolutions)
+        if len(resolutions) == 0:
+            print("WARNING: No EyefulTower resolutions specified. Nothing will be downloaded.")
+
+        driver = awscli.clidriver.create_clidriver()
+
+        for i, capture in enumerate(captures):
+            base_url = f"s3://fb-baas-f32eacb9-8abb-11eb-b2b8-4857dd089e15/EyefulTower/{capture}/"
```

Comment:
What is someone puts a capture name in that doesn't exist? You might want to catch this?

Reply:
There's input validation performed by the tyro library to catch this:

```
(nerfstudio12) PS C:\Users\vasuagrawal\Documents\nerfstudio> python .\nerfstudio\scripts\downloads\download_data.py eyefultower --capture-name table foo
╭─ Parsing error ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Argument --capture-name: invalid choice: 'foo' (choose from 'apartment', 'kitchen', 'workshop', 'office_view2', 'office2', 'table', 'all', 'office1b', 'riverview', 'office_view1', 'office1a', 'seating_area') │
│ ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── │
│ For full helptext, run download_data.py eyefultower --help                                                                                                                                                      │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

Reply:
Ah yes great!